lightning/src/ln/msgs.rs

```diff
@@ -305,6 +305,58 @@ impl NetAddress {
 	}
 }
 
+/// A "set" of addresses which enforces that there can be only up to one of each net address type.
```

Comment:
how does this actually enforce that the underlying NetAddress objects behave are what's expected?

Reply:
Because the public API only allows you to add one address per type (replacing the previous one if there is one already).

Reply:
see the set_address methods now. Also, hadn't realized NetAddress wasn't a standard library.